File: content-solutions/heap/merge-piles.py
import heapq


def solution(piles):
    h = piles[:]
    heapq.heapify(h)
    total = 0
    while len(h) >= 2:
        a = heapq.heappop(h)
        b = heapq.heappop(h)
        total += a + b
        heapq.heappush(h, a + b)
    return total

File: content-solutions/heap/merge-piles.java
import java.util.*;

class Solution {
    public long solution(int[] piles) {
        PriorityQueue<Long> h = new PriorityQueue<>();
        for (int p : piles) h.offer((long) p);
        long total = 0;
        while (h.size() >= 2) {
            long a = h.poll(), b = h.poll();
            total += a + b;
            h.offer(a + b);
        }
        return total;
    }
}
